packages/streamline/src/components/list-items/list-item-group/list-item-group.tsx

```diff
@@ -18,9 +20,15 @@ export const ListItemGroup = ({
         </Text>
       ) : null}
       <AnimatedBox borderRadius="lg" overflow="hidden">
-        {items.map((props, index) => (
-          <ListItem key={index} {...props} showDivider={index !== lastIndex} />
-        ))}
+        {items.map((props, index) => {
+          const showDivider = index !== lastIndex;
+          if (isListItemValueProps(props)) {
+            return (
+              <ListItemValue key={index} {...props} showDivider={showDivider} />
+            );
+          }
+          return <ListItem key={index} {...props} showDivider={showDivider} />;
```

Comment:
we can extract the pattern matching logic inside a special react component 

```
// Define the components for each user type
const AdminComponent: React.FC = () => <div>Admin Component</div>;
const ManagerComponent: React.FC = () => <div>Manager Component</div>;
const UserComponent: React.FC = () => <div>User Component</div>;

// Define the main component
const UserComponent: React.FC<UserComponentProps> = ({ user }) => {
  // Use pattern matching with switch case
  switch (user.type) {
    case 'admin':
      return <AdminComponent />;
    case 'manager':
      return <ManagerComponent />;
    case 'user':
      return <UserComponent />;
    default:
      return null;
  }
};
```

we can have more advance case with a switch(true) ...

```
// Define the main component
const UserComponent: React.FC<UserComponentProps> = ({ user }) => {
  // Use pattern matching with switch case
  switch (true) {
    case checkAdmin() :
      return <AdminComponent />;
    case checkManager():
      return <ManagerComponent />;
    case checkUser():
      return <UserComponent />;
    default:
      return null;
  }
};
```